crates/provider-sdk/src/lib.rs

```diff
@@ -142,3 +150,107 @@ pub trait ProviderHandler: Sync {
     /// Handle system shutdown message
     async fn shutdown(&self) {}
 }
+
+/// Parse a `wit_parser::Resolve` from anything that implements a mechanism for looking up file content
+async fn parse_wit_from_file_content_lookup(
+    _lookup: Box<dyn FileLookup + Send>,
+) -> ProviderInitResult<Resolve> {
+    // TODO: use a file lookup to build a Resolve
+    Ok(Resolve::default())
+}
+
+/// Generate a list of incoming NATS RPC topics on which the provider should listen
+fn generate_incoming_rpcs_topics(
+    lattice_name: impl AsRef<str>,
+    wrpc_version: impl AsRef<str>,
+    resolve: &Resolve,
+) -> ProviderInitResult<Vec<String>> {
+    let lattice_name = lattice_name.as_ref();
+    let wrpc_version = wrpc_version.as_ref();
+
+    let mut topics = Vec::new();
+    for (_, world) in resolve.worlds.iter() {
+        let world_name = &world.name;
+
+        let pkg_id = world.package.ok_or_else(|| {
+            ProviderInitError::Initialization(format!(
+                "unexpectedly missing package on with world [{world_name}]"
+            ))
+        })?;
+        let pkg = resolve.packages.get(pkg_id).ok_or_else(|| {
+            ProviderInitError::Initialization(format!(
+                "unexpectedly package with ID [{pkg_id:?}] in world [{world_name}]"
+            ))
+        })?;
+        let pkg_name = &pkg.name.name;
+        let ns = &pkg.name.namespace;
+
+        for (world_item, _) in world.exports.iter() {
+            match world_item {
+                wit_parser::WorldKey::Name(_) => continue,
+                wit_parser::WorldKey::Interface(iface_id) => {
+                    if let Some(iface) = resolve.interfaces.get(*iface_id) {
+                        let iface_name = iface.name.as_ref().ok_or_else(|| {
+                            ProviderInitError::Initialization(format!(
+                                "unexpectedly un-named iface ID [{iface_id:?}] in world [{world_name}]"
+                            ))
+                        })?;
+
+                        for (fn_name, _) in iface.functions.iter() {
+                            topics.push(format!(
+                                "{lattice_name}.wrpc.{wrpc_version}.{ns}:${pkg_name}/{iface_name}.{fn_name}"
```

Comment:
Just a heads up that we are going to need to escape version identifiers with `!`. So `0.1.0` -> `0!1!0` since we use the `.` char for topic splitting and matching. Should have @rvolosatovs or @brooksmtownsend confirm this before changing though

Reply:
Yup, I think you're right on this -- it's [in here](https://github.com/wasmCloud/wasmCloud/issues/1389), so I'll change this

Reply:
We're not doing that for 0.0.1

Reply:
welp, [evidently not](https://github.com/wasmCloud/wasmCloud/pull/1469#discussion_r1481507649)